wrappers/src/fdw/cognito_fdw/cognito_client/rows_iterator.rs

```
@@ -0,0 +1,86 @@
+use crate::fdw::cognito_fdw::cognito_client::CognitoClientError;
+
+use crate::fdw::cognito_fdw::cognito_client::CreateRuntimeError;
+
+use std::collections::VecDeque;
+use supabase_wrappers::prelude::{Column, Row};
+
+pub(crate) struct RowsIterator {
+    cognito_client: aws_sdk_cognitoidentityprovider::Client,
+    columns: Vec<Column>,
+    rows: VecDeque<Row>,
+    user_pool_id: String,
+    have_more_rows: bool,
+    pagination_token: Option<String>,
+}
+
+impl RowsIterator {
+    pub(crate) fn new(
+        columns: Vec<Column>,
+        user_pool_id: String,
+        cognito_client: aws_sdk_cognitoidentityprovider::Client,
+    ) -> Self {
+        Self {
+            columns,
+            cognito_client,
+            user_pool_id,
+            rows: VecDeque::new(),
+            have_more_rows: true,
+            pagination_token: None,
+        }
+    }
+
+    fn fetch_rows_batch(&mut self) -> Result<Option<Row>, CognitoClientError> {
+        self.have_more_rows = false;
+        let rt = tokio::runtime::Runtime::new()
+            .map_err(CreateRuntimeError::FailedToCreateAsyncRuntime)?;
+
+        let mut request = self
+            .cognito_client
+            .list_users()
+            .user_pool_id(self.user_pool_id.clone());
+
+        if let Some(ref token) = self.pagination_token {
+            request = request.pagination_token(token.clone());
+        }
+        self.rows = rt.block_on(async {
+            match request.send().await {
+                Ok(response) => {
+                    self.pagination_token = response.pagination_token.clone();
+                    response
+                        .users
+                        .clone()
+                        .unwrap_or_else(Vec::new)
+                        .into_iter()
+                        .map(|u| u.into_row(&self.columns))
+                        .collect::<VecDeque<Row>>()
+                }
+                Err(_) => {
```

Return an error here instead of an empty `VecDeque`?